lib/screens/user/blood_donation_reservation_page.dart

```diff
@@ -62,29 +63,27 @@ class _BloodDonationReservationPageState
 
   void _submitForm() async {
     if (_formKey.currentState!.validate() && _selectedBloodType != null) {
-      await FirebaseFirestore.instance.collection('donation_date').add({
-        'donor_name': _nameController.text,
-        'email': _emailController.text,
-        'date': _dateController.text,
-        'blood_type': _selectedBloodType,
-        'user_id': _userId
-      });
+      await _donationService.saveBloodDonationData(
+        donorName: _nameController.text,
+        email: _emailController.text,
+        date: _dateController.text,
+        bloodType: _selectedBloodType!,
+        userId: _userId!,
+      );
 
-      // Clear form data and reset blood type dropdown
```

Comment:
please remove all comments like these. 

Backstory: 
It's unnecessary to use comments for things that are obvious. 

> Don’t over-comment. Too many comments can make code harder to read and understand. Only add comments when necessary, not to every line of code. Avoid redundant comments.